### PR TITLE
Config templates

### PR DESCRIPTION
**Pull request by liamsi**
To see the changes made to the config I've used for my test-sentry, have a look at this commit: https://github.com/mmmh-studio/validator/commit/81dd43bf61fba1df1f884f65c9641facbd0612c2

Note:
- templates can be supposedly be transformed into actual gaia config via: https://www.terraform.io/docs/configuration/functions/templatefile.html
- string templates according to: https://www.terraform.io/docs/configuration/expressions.html#string-templates
- config generated via gaiad version `2.0.3`

related reading: https://forum.cosmos.network/t/sentry-node-architecture-overview/454

### CODE REVIEW

File: config-templates/sentry/config/config.toml [hunk 1]
```diff
@@ -158,10 +159,14 @@ laddr = "tcp://0.0.0.0:26656"
 external_address = ""
 
 # Comma separated list of seed nodes to connect to
-seeds = "344e7c6cf62d26a635cc668e6dfbead96c7e0f43@52.57.29.196:26656,352bc356eabc1b908bef70cd6e644b6dd44923cb@35.158.255.161:26656,3e16af0cead27979e1fc3dac57d03df3c7a77acc@3.87.179.235:26656,ba3bacc714817218562f743178228f23678b2873@public-seed-node.cosmoshub.certus.one:26656,2626942148fd39830cb7a3acccb235fab0332d86@173.212.199.36:26656,3028c6ee9be21f0d34be3e97a59b093e15ec0658@91.205.173.168:26656,89e4b72625c0a13d6f62e3cd9d40bfc444cbfa77@34.65.6.52:26656"
+# example that worked on the hub-3 mainnet:
+# seeds = "344e7c6cf62d26a635cc668e6dfbead96c7e0f43@52.57.29.196:26656,352bc356eabc1b908bef70cd6e644b6dd44923cb@35.158.255.161:26656,3e16af0cead27979e1fc3dac57d03df3c7a77acc@3.87.179.235:26656,ba3bacc714817218562f743178228f23678b2873@public-seed-node.cosmoshub.certus.one:26656,2626942148fd39830cb7a3acccb235fab0332d86@173.212.199.36:26656,3028c6ee9be21f0d34be3e97a59b093e15ec0658@91.205.173.168:26656,89e4b72625c0a13d6f62e3cd9d40bfc444cbfa77@34.65.6.52:26656"
+seeds = "&{sentry.seed_list}"
```
Comment (liamsi) [inline config-templates/sentry/config/config.toml]:
```suggestion
seeds = "${sentry.seed_list}"
```

**Comment by liamsi** [comment]
Following vars need to exist:

Sentry:
 - sentry.moniker
 - sentry.seed_list
 - sentry.validator_node_id
 - sentry.validator_node_ip
 - sentry.validator_node_port

validator:
 - validator.moniker
 - validator.sentries
 - validator.kms_addr

File: config-templates/kms/tmkms.toml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+# Note(Ismail):
+# could be e.g. "tcp://f88883b673fc69d7869cab098de3bafc2ff76eb8@example1.example.com:26658"
+# or or addr = "unix:///path/to/socket"
+# !!Important!! to keep in mind: "the Secret Connection key associated with gaiad's priv_validator_laddr
+# is randomly generated each time it starts. It is static so long as gaiad is running,
+# but changes the next time you start it."
+# Details:
+# https://github.com/tendermint/kms/issues/371
+# https://github.com/tendermint/tendermint/issues/3105
+addr = "${kms.validator_addr}"
```
Comment (Liamsi) [inline config-templates/kms/tmkms.toml]: This is particularly relevant.

Comment (liamsi) [inline config-templates/kms/tmkms.toml:45]: This is particularly relevant.

File: config-templates/validator/config/config.toml [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+# The path to a file containing matching private key that is used to create the HTTPS server.
+# Migth be either absolute path or path related to tendermint's config directory.
+# NOTE: both tls_cert_file and tls_key_file must be present for Tendermint to create HTTPS server. Otherwise, HTTP server is run.
+tls_key_file = ""
+
+##### peer to peer configuration options #####
+[p2p]
+
+# Address to listen for incoming connections
+laddr = "tcp://0.0.0.0:26656"
```
Comment (xla) [inline config-templates/validator/config/config.toml]: Let's explicitly configure NIC addresses for heighten securtiy.

File: config-templates/validator/config/config.toml [hunk 4]
```diff
@@ -0,0 +1,316 @@
+# This is a TOML config file.
+# For more information, see https://github.com/toml-lang/toml
+
+##### main base config options #####
+
+# TCP or UNIX socket address of the ABCI application,
+# or the name of an ABCI application compiled in with the Tendermint binary
+proxy_app = "tcp://127.0.0.1:26658"
```
Comment (xla) [inline config-templates/validator/config/config.toml]: Should be app name for `gaiad` use-case.

File: config-templates/validator/config/config.toml [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+#   - requires gcc
+#   - use cleveldb build tag (go build -tags cleveldb)
+# * boltdb (uses etcd's fork of bolt - github.com/etcd-io/bbolt)
+#   - EXPERIMENTAL
+#   - may be faster is some use-cases (random reads - indexer)
+#   - use boltdb build tag (go build -tags boltdb)
+db_backend = "goleveldb"
+
+# Database directory
+db_dir = "data"
```
Comment (xla) [inline config-templates/validator/config/config.toml]: Should be configurable for flexibility with virtual storage.

File: config-templates/validator/config/config.toml [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+
+# Output level for logging, including package level options
+log_level = "main:info,state:info,*:error"
+
+# Output format: 'plain' (colored text) or 'json'
+log_format = "plain"
```
Comment (xla) [inline config-templates/validator/config/config.toml]: Should be `json` for easier consumption in log infra.

File: config-templates/validator/config/config.toml [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+
+##### additional base config options #####
+
+# Path to the JSON file containing the initial validator set and other meta data
+genesis_file = "config/genesis.json"
```
Comment (xla) [inline config-templates/validator/config/config.toml]: Config dir should be configurable and the path used here as var so we can store them and write them from the automation.

File: config-templates/validator/config/config.toml [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+
+# Path to the JSON file containing the private key to use as a validator in the consensus protocol
+priv_validator_key_file = "config/priv_validator_key.json"
```
Comment (xla) [inline config-templates/validator/config/config.toml]: Config dir.

File: config-templates/validator/config/config.toml [hunk 9]
```diff
@@ -0,0 +1,316 @@
...
+
+# Path to the JSON file containing the last sign state of a validator
+priv_validator_state_file = "data/priv_validator_state.json"
```
Comment (xla) [inline config-templates/validator/config/config.toml]: Data dir configurable.

File: config-templates/validator/config/config.toml [hunk 10]
```diff
@@ -0,0 +1,316 @@
...
+
+# TCP or UNIX socket address for Tendermint to listen on for
+# connections from an external PrivValidator process
+priv_validator_laddr = "${validator.kms_addr}"
+
+# Path to the JSON file containing the private key to use for node authentication in the p2p protocol
+node_key_file = "config/node_key.json"
```
Comment (xla) [inline config-templates/validator/config/config.toml]: Config dir configurable.

File: config-templates/validator/config/config.toml [hunk 11]
```diff
@@ -0,0 +1,316 @@
...
+
+# Mechanism to connect to the ABCI application: socket | grpc
+abci = "socket"
+
+# TCP or UNIX socket address for the profiling server to listen on
+prof_laddr = "localhost:6060"
```
Comment (xla) [inline config-templates/validator/config/config.toml]: Is this metrics? Should also be configurable.

File: config-templates/validator/config/config.toml [hunk 12]
```diff
@@ -0,0 +1,316 @@
...
+# so the app can decide if we should keep the connection or not
+filter_peers = false
+
+##### advanced configuration options #####
+
+##### rpc server configuration options #####
+[rpc]
+
+# TCP or UNIX socket address for the RPC server to listen on
+laddr = "tcp://127.0.0.1:26657"
```
Comment (xla) [inline config-templates/validator/config/config.toml]: Let's turn off rpc.

Comment (xla) [reply config-templates/validator/config/config.toml]: Alternatively make it configurable so we can bind it to the utility network.

**Review by xla — CHANGES_REQUESTED** [review]
Dope!

Let's consolidate validator and sentry TOMLs and decide via the variables if it used as as either.

**Comment by liamsi** [comment]
All addressed in 
aab6170

**Review by xla — COMMENTED** [review]
Dope! Let's merge that bad boi.

📰 🍂 🌊 💻

**Review by xla — APPROVED** [review]

**Comment by liamsi** [comment]
Thanks @xla

**Pull request merged**